### PR TITLE
[Ahuna] Update Assistant Details UX

### LINKED ISSUE #3265: Update Assistant Detailed View
As per [here](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4050-22778&mode=design&t=wGVRT1IiZNJ18Mm1-4) and note the 4 other views

The screen depends on 2 settings:

- The member rights (Member, Builder)
- The assistant type (Default, Company, Shared, Personal)

Edit rights cover editing Type (Company, Shared, Personal) and editing the assistant instructions.

All actions are in a dropdown behind an Action menu.

Note: use the newly updated dropdown component that can do all that's in the figma

Also, there is a specific interaction to add/remove from list (see section Assistant Detail view of the [issue](https://github.com/dust-tt/decisions/issues/147#issue-2081773564))
For this, a specific (non-sparkle) component should be made and mutually used with the assistant preview in the gallery

Full [figma](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4050-22251&mode=design&t=Dlhq25QyeMahW2OI-0) (source of truth)

Full [issue](https://github.com/dust-tt/decisions/issues/147#issue-2081773564)

### PR DESCRIPTION
**Pull request by philipperolet**
## Description
Fixes #3265 for the most. See below 2 elements that will follow up shortly

### Follow-up work
- add archive button to menu
- add try button to menu

### TBD Duncid
- Add/Remove button for assitant -> is it necessary since add/remove is already in the menu?
- Modal closing -> should we use a left chevron rather than a cross? (since cross is used everywhere else in that kind of modal)

image

## Risks
UX risks. Deployed on front-edge to test

### CODE REVIEW

File: front/components/assistant_builder/Sharing.tsx [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+
+type ConfirmationModalDataType = {
+  title: string;
+  text: string;
+  confirmText: string;
+  showUsage?: boolean;
+  variant: "primary" | "primaryWarning";
+};
+
+const SCOPE_INFO: Record<
```
Comment (flvndvd) [inline front/components/assistant_builder/Sharing.tsx:27]: I'll move this to a dedicated component so we can avoid duplicating this logic everywhere.

Comment (philipperolet) [reply front/components/assistant_builder/Sharing.tsx:27]: Agreed there should be an `export` here
IMHO this is the dedicated lib of components (Sharing); when it becomes too large we could indeed split into front & lib components

Comment (flvndvd) [reply front/components/assistant_builder/Sharing.tsx:27]: Seen IRL, this comment was meant for the Chip logic that will used both in the AssistantPreview and the assistant builder logic.

Comment (philipperolet) [reply front/components/assistant_builder/Sharing.tsx:27]: As discussed will extract a SharingChip & move Sharing to another place

File: front/components/assistant/Sharing.tsx [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+  setNewScope,
+}: {
+  owner: WorkspaceType;
+  agentConfigurationId: string | null;
+  disabled?: boolean;
+  initialScope: AgentConfigurationScope;
+  newScope: AgentConfigurationScope;
+  setNewScope: (scope: Exclude<AgentConfigurationScope, "global">) => void;
+}) {
+  const { agentConfiguration } = useAgentConfiguration({
```
Comment (flvndvd) [inline front/components/assistant/Sharing.tsx:155]: Can we avoid an extra `useAgentConfiguration` here?

**Review by Duncid — APPROVED** [review]

File: front/components/assistant/Sharing.tsx [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+                    !agentConfigurationId ||
+                    // selection unchanged
+                    entryScope === newScope ||
+                    // selection back to initial state
+                    entryScope === initialScope ||
+                    // the only user of the assistant is the user changing the scope
+                    ((entryScope === "private" || entryScope === "company") &&
+                      assistantInMyList &&
+                      (!agentUsage.agentUsage ||
+                        agentUsage.agentUsage.userCount <= 1))
```
Comment (flvndvd) [inline front/components/assistant/Sharing.tsx:266]: Are we sure that this one user is the current user?

File: front/components/assistant/conversation/AssistantEditionMenu.tsx [hunk 4]
```diff
@@ -18,14 +19,20 @@ import { useAgentConfiguration } from "@app/lib/swr";
 interface AssistantEditionMenuProps {
   agentConfigurationId: string;
   owner: WorkspaceType;
+  variant: "button" | "plain";
 }
 
+AssistantEditionMenu.defaultProps = {
```
Comment (flvndvd) [inline front/components/assistant/conversation/AssistantEditionMenu.tsx:25]: 🙌

File: front/components/assistant/conversation/AssistantEditionMenu.tsx [hunk 5]
```diff
@@ -82,13 +89,31 @@ export function AssistantEditionMenu({
...
   };
 
+  const dropdownButton = (() => {
+    switch (variant) {
+      case "button":
+        return (
+          <Button
+            key="show_details"
+            icon={MoreIcon}
+            label={"Actions"}
```
Comment (flvndvd) [inline front/components/assistant/conversation/AssistantEditionMenu.tsx:99]: `nit`
```suggestion
            label="Actions"
```

**Review by flvndvd — COMMENTED** [review]
LGTM, left a few comments.

File: front/components/assistant/Sharing.tsx (hunk 3 above)
Comment (philipperolet) [reply front/components/assistant/Sharing.tsx:266]: Yes; there's a change to the usercount logic that the user count should be the number of users that have the assistant in their list. And above we check the assistant is in this user's list

But I changed it to === 1 which is clearer

Comment (philipperolet) [reply front/components/assistant/Sharing.tsx:266]: (the change is not active yet though but will be soon, cf ahuna cards)

File: front/components/assistant/Sharing.tsx (hunk 2 above)
Comment (philipperolet) [reply front/components/assistant/Sharing.tsx:155]: wdym? we need it below for name + inmylist, + if it's used by a parent component it's cached so not costly right?

Comment (philipperolet) [reply front/components/assistant/Sharing.tsx:155]: or would you pass an agentconfigobject directly?

**Review by flvndvd — APPROVED** [review]

**Pull request merged**